Compile Error on FreeBSD 12.2

i follow the guide lines for compile in the wiki:

when i run make i get this error:

Making all in tape_drivers/freebsd/cam
/bin/sh ../../../../libtool  --tag=CC --tag=disable-static  --mode=compile cc -DHAVE_CONFIG_H -I. -I../../../..  -D_GNU_SOURCE -I../../../../src -DLTFS_CONFIG_FILE='"/usr/local/etc/ltfs.conf"' -DLTFS_BASE_DIR='"/usr/local"' -I ../../.. -I ../..  -Wall -Wsign-compare -fsigned-char -I/usr/local/include/fuse -D_FILE_OFFSET_BITS=64 -I/usr/local/include/uuid -I/usr/local/include -I/usr/local/include/libxml2 -DU_DISABLE_RENAMING=1 -I/usr/local/include -D ENABLE_SNMP -O2 -pipe -I/usr/local/include -I/include -fstack-protector-strong -fno-strict-aliasing -std=c99 -DNETSNMP_ENABLE_IPV6 -fno-strict-aliasing -DNETSNMP_REMOVE_U64 -O2 -pipe -I/usr/local/include -I/include -fstack-protector-strong -fno-strict-aliasing -std=c99 -Ufreebsd12 -Dfreebsd12=freebsd12 -DHAS_FPSETMASK -DHAS_FLOATINGPOINT_H -DUSE_THREAD_SAFE_LOCALE -fno-strict-aliasing -pipe -fstack-protector-strong -I/usr/local/include -I/usr/local/lib/perl5/5.32/mach/CORE -I/usr/include -I. -I/usr/local/include -g -O2 -D_FORTIFY_SOURCE=0 -O0 -ggdb -MT libtape_cam_la-cam_tc.lo -MD -MP -MF .deps/libtape_cam_la-cam_tc.Tpo -c -o libtape_cam_la-cam_tc.lo `test -f 'cam_tc.c' || echo './'`cam_tc.c
libtool: compile:  cc -DHAVE_CONFIG_H -I. -I../../../.. -D_GNU_SOURCE -I../../../../src -DLTFS_CONFIG_FILE=\"/usr/local/etc/ltfs.conf\" -DLTFS_BASE_DIR=\"/usr/local\" -I ../../.. -I ../.. -Wall -Wsign-compare -fsigned-char -I/usr/local/include/fuse -D_FILE_OFFSET_BITS=64 -I/usr/local/include/uuid -I/usr/local/include -I/usr/local/include/libxml2 -DU_DISABLE_RENAMING=1 -I/usr/local/include -D ENABLE_SNMP -O2 -pipe -I/usr/local/include -I/include -fstack-protector-strong -fno-strict-aliasing -std=c99 -DNETSNMP_ENABLE_IPV6 -fno-strict-aliasing -DNETSNMP_REMOVE_U64 -O2 -pipe -I/usr/local/include -I/include -fstack-protector-strong -fno-strict-aliasing -std=c99 -Ufreebsd12 -Dfreebsd12=freebsd12 -DHAS_FPSETMASK -DHAS_FLOATINGPOINT_H -DUSE_THREAD_SAFE_LOCALE -fno-strict-aliasing -pipe -fstack-protector-strong -I/usr/local/include -I/usr/local/lib/perl5/5.32/mach/CORE -I/usr/include -I. -I/usr/local/include -g -O2 -D_FORTIFY_SOURCE=0 -O0 -ggdb -MT libtape_cam_la-cam_tc.lo -MD -MP -MF .deps/libtape_cam_la-cam_tc.Tpo -c cam_tc.c  -fPIC -DPIC -o .libs/libtape_cam_la-cam_tc.o
cam_tc.c:1142:39: error: use of undeclared identifier 'priv'
                softc->cart_type = assume_cart_type(priv->density_code);
                                                    ^
cam_tc.c:4126:28: warning: incompatible pointer types initializing 'int (*)(void *, const uint8_t, const uint8_t, unsigned char *, const size_t)' (aka 'int (*)(void *, const unsigned char, const
      unsigned char, unsigned char *, const unsigned long)') with an expression of type 'int (struct camtape_data *, const uint8_t, const uint8_t, unsigned char *, const size_t)' (aka 'int (struct
      camtape_data *, const unsigned char, const unsigned char, unsigned char *, const unsigned long)') [-Wincompatible-pointer-types]
        .logsense               = camtape_logsense,
                                  ^~~~~~~~~~~~~~~~
1 warning and 1 error generated.
*** Error code 1

Stop.
make[3]: stopped in /root/ltfs/src/tape_drivers/freebsd/cam
*** Error code 1
*** Error code 1
*** Error code 1


full compile paste here:    https://paste.sh/4J6jbVyj#4y9bjTOe_5v-f1UmOsu_Xt3k

--- a/src/tape_drivers/freebsd/cam/cam_cmn.h
+++ b/src/tape_drivers/freebsd/cam/cam_cmn.h
@@ -451,8 +451,8 @@ extern int camtape_get_info(void *device, struct tc_drive_info *info);
 extern int camtape_set_profiler(void *device, char *work_dir, bool enable);
 extern int camtape_get_timeout(struct timeout_tape *table, int op_code);
 
-extern int camtape_logsense_page(struct camtape_data *softc, const uint8_t page, const uint8_t subpage,
-								 unsigned char *buf, const size_t size);
+extern int camtape_logsense(void *device, const uint8_t page, const uint8_t subpage,
+							unsigned char *buf, const size_t size);
 
 /**
  * Get Dump
@@ -579,7 +579,7 @@ static inline bool is_dump_required_error(struct camtape_data *softc, int ret, b
 		log_page.parm_pointer = 0;
 		memset(log_page.data, 0, LOGSENSEPAGE);
 
-		rc = camtape_logsense_page(softc, log_page.page_code, 0, (unsigned char *)&log_page.data,
+		rc = camtape_logsense(softc, log_page.page_code, 0, (unsigned char *)&log_page.data,
 								   LOGSENSEPAGE);
 
 		ans = (rc == DEVICE_GOOD);

--- a/src/tape_drivers/freebsd/cam/cam_tc.c
+++ b/src/tape_drivers/freebsd/cam/cam_tc.c
@@ -1139,7 +1139,7 @@ int camtape_load(void *device, struct tc_position *pos)
 		/*
 		 * Non-IBM drive doesn't have cartridge type so need to assume from density code.
 		 */
-		softc->cart_type = assume_cart_type(priv->density_code);
+		softc->cart_type = assume_cart_type(softc->density_code);
 		if (buf[2] == 0x01)
 			softc->is_worm = true;
 	} else {
@@ -1686,7 +1686,7 @@ bailout:
  */
 #define MAX_UINT16 (0x0000FFFF)
 
-int camtape_logsense(struct camtape_data *softc, const uint8_t page, const uint8_t subpage,
+int camtape_logsense(void *device, const uint8_t page, const uint8_t subpage,
 					 unsigned char *buf, const size_t size)
 {
 	int rc = DEVICE_GOOD;
@@ -1694,6 +1694,8 @@ int camtape_logsense(struct camtape_data *softc, const uint8_t page, const uint8
 	struct scsi_log_sense *scsi_cmd;
 	union ccb *ccb = NULL;
 	int timeout;
+
+	struct camtape_data *softc = device;
 
 	unsigned int len = 0;
 	unsigned char *inner_buf = NULL;

